airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/state/PipelineEventBookkeepingRouter.kt

```diff
@@ -216,7 +216,9 @@ class PipelineEventBookkeepingRouter(
                 catalog.streams.forEach {
                     val sawComplete = sawEndOfStreamComplete.contains(it.descriptor)
                     val manager = syncManager.getStreamManager(it.descriptor)
-                    manager.markEndOfStream(sawComplete)
+                    if(sawComplete) {
+                        manager.markEndOfStream(sawComplete)
+                    }
```

Comment:
Not sure why this is a bug or how this fixes things, but it seems fine?

Reply:
@tryangul I'm not 100% sure either, but in debugging, I noticed that the method gets called twice from two different threads.  As a result, the last thread wins, which is actually the first in terms of execution (it tries to set the value to false).  Given that the underlying boolean is initialized to false, the only reason to update this is once we have seen a completed value, regardless of the order of the threads.  I think that is why this works.

Reply:
That makes sense, but it feels like the real bug is calling this twice. Debugging that is probably a whole rabbit hole, so this seems like a reasonable fix. I will say as a third party coming across this, I would not understand this if statement at all. It probably is worth adding a comment.